src/OrasProject.Oras/Extensions.cs

```diff
@@ -13,14 +13,40 @@
 
 using OrasProject.Oras.Oci;
 using System;
+using System.Collections.Generic;
+using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
-using static OrasProject.Oras.Content.Extensions;
+using OrasProject.Oras.Content;
+using OrasProject.Oras.Registry;
 
 namespace OrasProject.Oras;
 
 public static class Extensions
 {
+    
+    private static readonly SemaphoreSlim _semaphore = new(1, 10);
+
+    /// <summary>
+    /// Proxy class is to cache the manifest for OCI image/index manifest to improve performance
+    /// </summary>
+    public class Proxy : IFetchable
+    {
+        public required IStorage MemoryStorage { get; set; }
+        public required ITarget Target { get; set; }
```

Comment:
The naming here is not self-explaining. Do you mean something like
```suggestion
        public required IReadOnlyStorage Source { get; init; }
        public required IStorage Cache { get; init; }
```